### PR TITLE
feat: add localization support

### PR DESCRIPTION
**Pull request by mahali-asmah**

### CODE REVIEW

File: public/locales/de/landing.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "title": "Dies ist eine Einstiegsvorlagee"
```
Comment (janhesters) [inline public/locales/de/landing.json:2]: Lol, let's go with "Dies ist ein Starter Template"

File: src/features/landing/landing-page-component.test.tsx [hunk 2]
```diff
@@ -6,11 +6,33 @@ import { render } from '@/test/tests-utils';
 import { LandingPageComponent } from './landing-page-component';
 
 describe('Landing page component', () => {
-  it('should render the landing page component', () => {
-    render(<LandingPageComponent />);
+  it('should render the landing page component', async () => {
+    render(await LandingPageComponent({ params: { lng: 'en' } }));
```
Comment (janhesters) [inline src/features/landing/landing-page-component.test.tsx:10]: Never seen this, is this how you render async components? 🤯

File: src/features/localization/server.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { createInstance } from 'i18next';
```
Comment (janhesters) [inline src/features/localization/server.ts:1]: Any reason we're doing it this way vs. [how the docs describe it](https://nextjs.org/docs/app/building-your-application/routing/internationalization#localization) with the `getDictionary` function?

Comment (mahali-asmah) [reply src/features/localization/server.ts:1]: I wanted to get it working from the server like we do with `releaselog`

**Pull request merged**